### PR TITLE
fix: Fixes various issue in the web app deployment script

### LINKED ISSUE #4049: Issues with azure webapp deployment script
## Describe the bug

The azure webapp deployment script has various issues e.g. bf luis:build has wrong --dialog argument. 
appsettings.deployment.json will never be used by the published bot and exposes local secrets to the remote folder. 

I'll prepare a pull request to fix these issues.

Best wishes,
Christian

## Version
Current main branch build

### PR DESCRIPTION
**Pull request by ltwlf**
## Description
The azure webapp deployment script has various issues e.g. bf luis:build has wrong --dialog argument.
appsettings.deployment.json will never be used by the published bot and exposes local secrets to the remote folder.

## Task Item
fixes #4049

### CODE REVIEW

File: runtime/dotnet/azurewebapp/Scripts/deploy.ps1 [hunk 1]
```diff
@@ -63,37 +63,46 @@ dotnet user-secrets init
...
 
-
 # Copy bot files to running folder
 $remoteBotPath = $(Join-Path $publishFolder "ComposerDialogs")
 Remove-Item $remoteBotPath -Recurse -ErrorAction Ignore
 
-
 if (-not $botPath) {
 	# If don't provide bot path, then try to copy all dialogs except the runtime folder in parent folder to the publishing folder (bin\Realse\ Folder)
-	$botPath = '..'
```
Comment (ltwlf) [inline runtime/dotnet/azurewebapp/Scripts/deploy.ps1:74]: Wrong default path. Must be on level higher.

File: runtime/dotnet/azurewebapp/Scripts/deploy.ps1 [hunk 2]
```diff
@@ -63,37 +63,46 @@ dotnet user-secrets init
...
+	$botPath = '../..'
 }
 
 $botPath = $(Join-Path $botPath '*')
 Write-Host "Publishing dialogs from external bot project: $($botPath)"
 Copy-Item -Path (Get-Item -Path $botPath -Exclude ('runtime', 'generated')).FullName -Destination $remoteBotPath -Recurse -Force -Container
 
 # Try to get luis config from appsettings
-$settings = Get-Content $(Join-Path $projFolder appsettings.deployment.json) | ConvertFrom-Json
```
Comment (ltwlf) [inline runtime/dotnet/azurewebapp/Scripts/deploy.ps1:82]: appsettings.deployment.json has no effect to the published bot, should be appsettings.json

File: runtime/dotnet/azurewebapp/Scripts/deploy.ps1 [hunk 3]
```diff
@@ -63,37 +63,46 @@ dotnet user-secrets init
...
+$settingsPath = $(Join-Path $remoteBotPath settings appsettings.json)
+$settings = Get-Content $settingsPath | ConvertFrom-Json
 $luisSettings = $settings.luis
 
 if (-not $luisAuthoringKey) {
 	$luisAuthoringKey = $luisSettings.authoringKey
 }
 
-if (-not $luisEndpointKey) {
```
Comment (ltwlf) [inline runtime/dotnet/azurewebapp/Scripts/deploy.ps1:89]: Secrets/key should not be published to the remote host -> should be configured via azure setting: luis__endpointkey

File: runtime/dotnet/azurewebapp/Scripts/deploy.ps1 [hunk 4]
```diff
@@ -63,37 +63,46 @@ dotnet user-secrets init
...
-	$luisEndpointKey = $luisSettings.endpointKey
-}
-
 if (-not $luisAuthoringRegion) {
 	$luisAuthoringRegion = $luisSettings.region
 }
 
+# set feature configuration
```
Comment (ltwlf) [inline runtime/dotnet/azurewebapp/Scripts/deploy.ps1:92]: Use the local settings features if configured

File: runtime/dotnet/azurewebapp/Scripts/deploy.ps1 [hunk 5]
```diff
@@ -115,49 +124,40 @@ if ($luisAuthoringKey -and $luisAuthoringRegion) {
...
-		$customizedSettings = Get-Content $(Join-Path $remoteBotPath settings appsettings.json) | ConvertFrom-Json
-		$customizedEnv = $customizedSettings.luis.environment
-		
+
 		# create generated folder if not exists
 		if (!(Test-Path generated)) {
 			New-Item -ItemType Directory -Force -Path generated
 		}
-		
-		bf luis:build --luConfig $(Join-Path $remoteBotPath luconfig.json) --botName $name --authoringKey $luisAuthoringKey --dialog --out .\generated --suffix $customizedEnv -f --region $luisAuthoringRegion
```
Comment (ltwlf) [inline runtime/dotnet/azurewebapp/Scripts/deploy.ps1:134]: the argument --dialog without value causes an error. In the latest branch it mus be crosstrained

File: runtime/dotnet/azurewebapp/Scripts/deploy.ps1 [hunk 6]
```diff
@@ -115,49 +124,40 @@ if ($luisAuthoringKey -and $luisAuthoringRegion) {
...
+
+		bf luis:build --luConfig $(Join-Path $remoteBotPath luconfig.json) --botName $name --authoringKey $luisAuthoringKey --dialog crosstrained  --out .\generated --suffix $environment -f --region $luisAuthoringRegion
 	}
 	else {
 		Write-Host "bf luis:build does not exist, use the following command to install:"
```
Comment (ltwlf) [inline runtime/dotnet/azurewebapp/Scripts/deploy.ps1:144]: Meanwhile there is a published version of the CLI

File: runtime/dotnet/azurewebapp/Scripts/deploy.ps1 [hunk 7]
```diff
@@ -226,27 +225,12 @@ if ($luisAuthoringKey -and $luisAuthoringRegion) {
...
-if (Test-Path $(Join-Path $publishFolder appsettings.deployment.json)) {
-	$settings = Get-Content $(Join-Path $publishFolder appsettings.deployment.json) | ConvertFrom-Json
-}
-else {
-	$settings = New-Object PSObject
-}
-
 $settings | Add-Member -Type NoteProperty -Force -Name 'feature' -Value $featureConfig
-$settings | ConvertTo-Json -depth 100 | Out-File $(Join-Path $publishFolder appsettings.deployment.json)
+$settings | ConvertTo-Json -depth 100 | Out-File $settingsPath
```
Comment (ltwlf) [inline runtime/dotnet/azurewebapp/Scripts/deploy.ps1:229]: Replace settings in composer settings folder. The settings will only includes the luis app id mappings and feature configuration. All other settings must be configured via the azure web app settings

**Comment by luhan2017** [comment]
@ltwlf , we are deprecating this scripts, and use the scripts generated in your bot folder instead, could you please try that one? sorry for the confusion.

**Comment by ltwlf** [comment]
I optimized the deploy script for CI/CD deployment and updated the readme accordingly. And I deleted the obsolete scripts. As soon this is merged I'm going to record a screen cast on how to do CI/CD pipelines  with composer projects.
@VanyLaw can you review the PR?

**Comment by cwhitten** [comment]
@ltwlf please update/rebase your branches to represent the appropriate diff. Many commits outside this change are now included and the surface area is much too large.

**Comment by ltwlf** [comment]
@cwhitten I rebased on the current main. The file changes are now correct again.

**Review by cwhitten — APPROVED** [review]

**Pull request merged**